src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1251,6 +1259,13 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 model_info_call_error = e  # save error to reraise it if model is not cached locally
 
         if not local_files_only:
+            filenames = {sibling.rfilename for sibling in info.siblings}
+            if variant is not None and _check_legacy_sharding_variant_format(filenames=filenames, variant=variant):
+                warn_msg = f"This serialization format is now deprecated to standardize the serialization format between `transformers` and `diffusers`. We recommend you to remove the existing files associated with the current variant ({variant}) and re-obtain them by running a `save_pretrained()`."
```

Comment:
```suggestion
warn_msg = (
    f"Warning: The repository contains sharded checkpoints for variant '{variant}' that may use a deprecated format. "
    "Please check your files carefully:\n\n"
    "- Correct format example: diffusion_pytorch_model.fp16-00003-of-00003.safetensors\n"
    "- Deprecated format example: diffusion_pytorch_model-00001-of-00002.fp16.safetensors\n\n"
    "If you find any files in the deprecated format:\n"
    "1. Remove all existing checkpoint files for this variant.\n"
    "2. Re-obtain the correct files by running `save_pretrained()`.\n\n"
    "This will ensure you're using the most up-to-date and compatible checkpoint format."
)
```